Add PLAYER_VS_PLAYER functionality to SDK and Controller

ballyalley-o opened the pull request:
This pull request adds the functionality to retrieve player vs player data to the SDK and Controller modules. It includes the following changes:

- Added the PLAYER_VS_PLAYER directory path to the SDK_DIR configuration object.

- Added the PLAYER_VS_PLAYER route to the SDK module.

- Updated the Franchise ownership field in the Player Module to be an array.

- Updated the properties of the Team model.

- Added the PLAYER_VS_PLAYER route to the SDK module.

- Updated the Team model properties.

- Created player base data in the SDKSetController.

- Created franchise history records based on the NBA API data in the SDKSetController.

- Added the getPlayerVsPlayer method to the SDKController to retrieve player vs player data.

- Added the getGameFinder method to the SDKController to retrieve game data.

These changes enhance the functionality of the application by providing a convenient way to access player vs player data and game data.

ballyalley-o committed:
SDK: Player vs Player #17: Add PLAYER_VS_PLAYER directory path to SDK_DIR

This commit adds the PLAYER_VS_PLAYER directory path to the SDK_DIR configuration object. The PLAYER_VS_PLAYER path is used to generate the URL for retrieving player vs player data. It takes two player IDs as parameters and returns the directory path for the specified players. This addition enhances the functionality of the application by providing a convenient way to access player vs player data.

ballyalley-o committed:
Controller: Player vs Player #18: Add PLAYER_VS_PLAYER directory path to SDK_DIR

This commit adds the PLAYER_VS_PLAYER directory path to the SDK_DIR configuration object. The PLAYER_VS_PLAYER path is used to generate the URL for retrieving player vs player data. It takes two player IDs as parameters and returns the directory path for the specified players. This addition enhances the functionality of the application by providing a convenient way to access player vs player data.

ballyalley-o committed:
Player Module #2: Update Franchise ownership field to be an array

ballyalley-o committed:
📃 docs(docs: Update Franchise ownership field to be an array):

ballyalley-o committed:
SDK: Player vs Player #17 & Controller: Player vs Player #18: Add PLAYER_VS_PLAYER route to SDK

This commit adds the PLAYER_VS_PLAYER route to the SDK module. The route is responsible for retrieving player vs player data by taking two player IDs as parameters and returning the directory path for the specified players. This addition enhances the functionality of the application by providing a convenient way to access player vs player data.

ballyalley-o committed:
SDK: Player vs Player #17: Add PLAYER_VS_PLAYER route to SDK module

This commit adds the PLAYER_VS_PLAYER route to the SDK module. The route allows for retrieving player vs player data by taking two player IDs as parameters and returning the directory path for the specified players. This addition enhances the functionality of the application by providing a convenient way to access player vs player data.

ballyalley-o committed:
refactor: Update Team model properties

The code changes in the `typings/model.d.ts` file update the properties of the `Team` model. The `name` property has been renamed to `teamName`, and new properties `win`, `loss`, `winPercentage`, `divRank`, `confRank`, `season`, `coach`, `franchise`, `retiredJerseys`, and `slug` have been added. These changes improve the accuracy and completeness of the `Team` model in the application.

in-a-iii commented on sdk/sdk-server.py:
```diff
@@ -80,6 +80,42 @@ def get_player_career(player_id):
+
+    """
+
+    season = request.args.get('season', '2023-24')
```
add all the params from sdk… make it available to the other end.

ballyalley-o replied:
will do.

ballyalley-o replied:
done. please review.

ballyalley-o committed:
feat: Add query parameters for player vs player data retrieval

This commit adds query parameters to the SDKController's getPlayerVsPlayerData method for retrieving player vs player data. The added query parameters include last_n_games, measure_type_detailed_defense, month, opponent_team_id, pace_adjust, per_mode_detailed, period, plus_minus, rank, season_type_playoffs, date_from_nullable, date_to_nullable, game_segment_nullable, league_id_nullable, location_nullable, outcome_nullable, season_segment_nullable, vs_conference_nullable, and vs_division_nullable. These query parameters enhance the functionality of the application by allowing more specific and detailed player vs player data retrieval.

ballyalley-o committed:
Add query parameters for player vs player data retrieval

in-a-iii reviewed (approved):
great work. good to go.
